sdk/go/telemetry/init.go

```diff
@@ -103,21 +104,104 @@ func ConfiguredSpanExporter(ctx context.Context) (sdktrace.SpanExporter, bool) {
 	return configuredSpanExporter, configuredSpanExporter != nil
 }
 
-func InitEmbedded(ctx context.Context, res *resource.Resource) context.Context {
-	traceCfg := Config{
-		Detect:   false, // false, since we want "live" exporting
-		Resource: res,
-	}
-	if exp, ok := ConfiguredSpanExporter(ctx); ok {
-		traceCfg.LiveTraceExporters = append(traceCfg.LiveTraceExporters, exp)
-	}
-	return Init(ctx, traceCfg)
+var configuredLogExporter sdklog.Exporter
+var configuredLogExporterOnce sync.Once
+
+func ConfiguredLogExporter(ctx context.Context) (sdklog.Exporter, bool) {
+	ctx = context.WithoutCancel(ctx)
+
+	configuredLogExporterOnce.Do(func() {
+		var err error
+
+		var endpoint string
+		if v := os.Getenv("OTEL_EXPORTER_OTLP_LOGS_ENDPOINT"); v != "" {
+			endpoint = v
+		} else if v := os.Getenv("OTEL_EXPORTER_OTLP_ENDPOINT"); v != "" {
+			// we can't assume all OTLP endpoints support logs. better to be explicit
+			// than have noisy otel errors.
+			return
+		}
+		if endpoint == "" {
+			return
+		}
+
+		var proto string
+		if v := os.Getenv("OTEL_EXPORTER_OTLP_LOGS_PROTOCOL"); v != "" {
+			proto = v
+		} else if v := os.Getenv("OTEL_EXPORTER_OTLP_PROTOCOL"); v != "" {
+			proto = v
+		} else {
+			// https://github.com/open-telemetry/opentelemetry-specification/blob/v1.8.0/specification/protocol/exporter.md#specify-protocol
+			proto = "http/protobuf"
+		}
+
+		switch proto {
+		case "http/protobuf", "http":
+			headers := map[string]string{}
+			if hs := os.Getenv("OTEL_EXPORTER_OTLP_HEADERS"); hs != "" {
+				for _, header := range strings.Split(hs, ",") {
+					name, value, _ := strings.Cut(header, "=")
+					headers[name] = value
+				}
+			}
+			configuredLogExporter, err = otlploghttp.New(ctx,
+				otlploghttp.WithEndpointURL(endpoint),
+				otlploghttp.WithHeaders(headers))
+
+		case "grpc":
+			// FIXME: bring back when it's actually implemented
```

Comment:
NOTE: currently-shipped Dagger uses gRPC for nested OTel logging, so this means a dev Dagger in shipped Dagger won't be able to emit logs.